src/network/networkd-dhcp-server.c

```diff
@@ -395,3 +404,169 @@ int config_parse_dhcp_server_emit(
                 emit->addresses[emit->n_addresses++] = a.in;
         }
 }
+
+int config_parse_dhcp_static_lease_address(
+        const char *unit,
+        const char *filename,
+        unsigned line,
+        const char *section,
+        unsigned section_line,
+        const char *lvalue,
+        int ltype,
+        const char *rvalue,
+        void *data,
+        void *userdata) {
+
+        Network *network = data;
+        sd_dhcp_static_lease **current_lease_ref = &network->current_static_lease;
+        sd_dhcp_static_lease *current_lease = NULL;
+        Hashmap **static_leases_ref = &network->dhcp_static_leases;
+        _cleanup_(sd_dhcp_static_lease_unrefp) sd_dhcp_static_lease *old = NULL;
+
+
+        union in_addr_union addr;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
```

Comment:
you assert that `data` (aka `network`) is non-NULL here, but further up you already dereferenced here. this is wrong....